Update project to make it work with CPC core changes

mcghorayeb opened the pull request:
Update the project to make it functional after the changes that were made to CPC core and other CPC libs:

- Remove butterknife
- Migrate androidX
- Change the TargetSDKVersion and CompileSDKVersion
- Change the repo location from artifact to nexus and change the gradle implementation versions for CPC libs

mcghorayeb committed:
Remove the butterknife library and replace its usage with the traditional findViewById

mcghorayeb committed:
Merge branch 'remove-butterknife' into update-project

mcghorayeb committed:
Do the appropriate changes to migrate to androidX

mcghorayeb committed:
Merge branch 'migrate-android-x' into update-project

mcghorayeb committed:
Change the repository location from artifactory to nexus

mcghorayeb committed:
Change the compile SDK version and the target SDK version from 27 to 28

mcghorayeb committed:
Upgrade the CPC included libraries to the latest working versions

mcghorayeb committed:
Update the README.md to include the repository location change

mcghorayeb committed:
Merge branch 'update-cpc-libraries' into update-project

mreynier commented on app/build.gradle:
```diff
@@ -47,12 +47,12 @@ android {
-    implementation "com.android.support:appcompat-v7:$support"
-    implementation "com.android.support:design:$support"
-    implementation 'com.android.support.constraint:constraint-layout:1.1.3'
+    implementation 'androidx.appcompat:appcompat:1.0.0'
```
pkoi ne pas garder la variable $support et mettre =1.0.0

mcghorayeb replied:
Oui tu as bien raison. Du coup c'est le boutton Migrate to AndroidX sur Android Studio qui m'a fait cela. Je fixe

mreynier commented on app/src/main/java/fr/coppernic/samples/pcsc/ui/MainActivity.java:
```diff
@@ -104,7 +98,33 @@ protected void onCreate(Bundle savedInstanceState) {
+        swConnect = findViewById(R.id.swConnect);
+        etApdu = findViewById(R.id.etApdu);
+
+        swConnect.setOnClickListener(this);
```
je prefere que tu mette new Onclicklistener ici plutot que le implements Onclicklistener en haut

mcghorayeb replied:
Okey, vu. Je le ferai avec plaisir

mcghorayeb committed:
Reuse the variable support for the dependencies that require the support version

--- This variable was removed by the automatic AndroidX migration of Android Studio

mcghorayeb committed:
Create a click listener for each view instead of implementing OnClickListener on the whole activity
